lib/rubocop/cop/style/exponential_notation.rb

```diff
@@ -60,8 +60,8 @@ module Style
       class ExponentialNotation < Base
         include ConfigurableEnforcedStyle
         MESSAGES = {
-          scientific: 'Use a mantissa in [1, 10[.',
-          engineering: 'Use an exponent divisible by 3 and a mantissa in [0.1, 1000[.',
+          scientific: 'Use a mantissa in (1...10).',
+          engineering: 'Use an exponent divisible by 3 and a mantissa in (0.1...1000).',
```

Comment:
I understand the reasoning, but regexp literals are not actually involved. As one option, using text seems less misleading, but the preference may vary. What do you think?

```suggestion
          scientific: 'Use a mantissa between 1 (inclusive) and 10 (exclusive).',
          engineering: 'Use an exponent divisible by 3 and a mantissa between 0.1 (inclusive) and 1000 (exclusive).',
```

Reply:
I find this overly wordy tbh, and probably still not as clear as possible. How about:
```suggestion
          scientific: 'Use a mantissa between 1 and 9, inclusive.',
          engineering: 'Use an exponent divisible by 3 and a mantissa between 0.1 and 999, inclusive.',
```

Reply:
Or, perhaps:

```suggestion
          scientific: 'Use a mantissa >= 1 and < 10.',
          engineering: 'Use an exponent divisible by 3 and a mantissa >= 0.1 and < 1000.',
```

Reply:
> Or, perhaps:

Yeah, I think this is clearer than the current state. Let's proceed with this message for now.

Reply:
> I find this overly wordy tbh, and probably still not as clear as possible.

Personally, I don’t think the wording is long. However, since exponential notations are probably not used frequently, an appropriate and clear term should be chosen for this rare detection :-)

Reply:
That suggestion looks fine to me as well. Personally, I don't use ranges often enough to be able to quickly tell if `..` or `...` is inclusive (and on which sides at that), so using `>=`  and `<` removes any ambiguity I might have if I would see such a message.